src/Confluent.Kafka/ConsumeException.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents an error that occured during message consumption.
+    /// </summary>
+    public class ConsumeException : Exception
```

Comment:
yes, it's in the `ConsumeRecord`. The only thing different from a non-error result is the type of key and value is `byte[]` not the type as per the generic parameter of `Consumer`.

Reply:
oh, i changed this so the non-error `ConsumeResult` doesn't contain an `Error` field, since doing so implies it might be something other than `NoError` (so people may feel obliged to check it). The exception is now build on KafkaException, which is what provides the error.